coveo-settings/tests_settings/test_validation.py

```diff
@@ -0,0 +1,38 @@
+import pytest
+
+from coveo_testing.markers import UnitTest
+from coveo_settings.settings import StringSetting, ValidationConfigurationError
+from coveo_settings.validation import InSequence
+
+
+@UnitTest
+def test_validation_in_sequence_positive() -> None:
+    """ Test InSequence validation """
+    validation = InSequence("first", "second")
+
+    # Positive test
+    test_setting_positive = StringSetting("ut", fallback="first", validation=validation)
+    assert test_setting_positive.value == "first"
+    assert str(test_setting_positive) == "first"
+    assert test_setting_positive.is_valid
+
+
+def test_validation_in_sequence_negative() -> None:
+    """ Test InSequence validation """
+    validation = InSequence("first", "second")
+
+    # Negative test
+    test_setting_negative = StringSetting("ut", fallback="third", validation=validation)
+    with pytest.raises(ValidationConfigurationError):
+        test_setting_negative.value
+    with pytest.raises(ValidationConfigurationError):
+        str(test_setting_negative)
+    assert not test_setting_negative.is_valid
+
+    # Test sensitive value
+    test_setting_sensitive = StringSetting(
+        "ut", sensitive=True, fallback="third", validation=validation
+    )
+    with pytest.raises(ValidationConfigurationError) as excinfo:
+        test_setting_sensitive.value
+    assert "third" not in str(excinfo.value)
```

Comment:
`with` doesn't scope the variable. Just like the last iteration of a `for x in something` will leave you with x:

```
>>> for x in [1, 2, 3]:
...     pass
>>> x == 3
True
```

Reply:
Interesting! That being said, since a typical use of `with` is to provide a resource that will be unavailable once the context is closed, for instance

```python
with open("/arbitrary.txt") as file:
    print(file.read())
```

I'd still move the `assert "third" not in str(excinfo.value)` inside the `with` block.

Reply:
@klalumiere No, that will not work. As soon as the exception occurs, the context manager leaves and that's when `excinfo` becomes available. A context manager cannot resume the execution of the block, it can only catch/reraise it, or specify a finally block.